addons/mail/models/res_users.py

```diff
@@ -35,13 +35,21 @@ class Users(models.Model):
         "Only internal user can receive notifications in Odoo",
     )]
 
-    @api.depends('share')
+    @api.depends('groups_id')
     def _compute_notification_type(self):
         for user in self:
-            # Only the internal users can receive notifications in Odoo
-            if user.share or not user.notification_type:
+            if user.has_group('mail.group_mail_notification_type_inbox'):
+                user.notification_type = 'inbox'
+            else:
                 user.notification_type = 'email'
```

Comment:
I wonder if it is really necessary (to check with @odoo/rd-sm). 🤔 🙂 
If yes, then I wonder if it should not be better to do the same logic done in `_compute_share` to avoid calling `has_group` method for each user in the loop. 🤔 
 https://github.com/odoo/odoo/blob/67323e4f0af942e60bc6332282315c811c7d7e24/odoo/addons/base/models/res_users.py#L428-L433

Reply:
I can definitely do the same logic than here, it is way performant when the field is recomputed for all users at once.

To the question
> I wonder if it is really necessary

What do you mean ? To just set the default, email, when the value is not set ?
This means if you set the group manually, the compute method will be called, but then you won't set the email/inbox according to the group. So I do think this is necessary.

Reply:
Indeed, I didn't think we could manually add the group. 🙂 

I just keep in mind the user can change the user type (internal user to portal user for instance). And so let's imagine the following case, the user is first an internal user and then the user becomes a portal user, shouldn't we have to remove the group to keep the behavior before your changes? That is, `notification_type` should be always set to `email` when the user is a portal one. 

What do you think? 🙂 🤔

Reply:
Note that it is already the case: notification_type is an editable computed fields based on share flag. When being set as a share user notification type is automatically updated to email.

Reply:
> Note that it is already the case: notification_type is an editable computed field based on share flag. When being set as a share user notification type is automatically updated to email.

When doing the changes, I wondered why this was implemented that way, as there was already a constrains checking portal users cannot have inbox notifications, and nothing broke on the runbot when I removed it.
With your explanation I now understand the functional requirements. I will change my code and add a unit test to test this functional requirement.